nomad/alloc_endpoint.go

```diff
@@ -29,6 +29,10 @@ func (a *Alloc) List(args *structs.AllocListRequest, reply *structs.AllocListRes
 	}
 	defer metrics.MeasureSince([]string{"nomad", "alloc", "list"}, time.Now())
 
+	if args.RequestNamespace() == structs.AllNamespacesSentinel {
+		return a.listAllNamespaces(args, reply)
+	}
```

Comment:
this early jump is the same pattern employed for `/v1/jobs` and `/v1/scaling/policies`:
https://github.com/hashicorp/nomad/blob/3926317613caae4f75f55c627ab08eb8aeb2a4d7/nomad/job_endpoint.go#L1301-L1303
https://github.com/hashicorp/nomad/blob/3926317613caae4f75f55c627ab08eb8aeb2a4d7/nomad/scaling_endpoint.go#L31-L33